### DISCUSSION TITLE
Use buf.build to download and manage dependencies

### DISCUSSION THREAD

**Discussion by trietsch** [devkanro/intellij-protobuf-plugin#214] [Q&A] [answered]
Hi @devkanro!

Thanks for this awesome plugin, which is much more feature rich than the built-in plugin from Jetbrains.
I noticed that this Protobuf plugin supports the option to invoke Buf CLI actions from the IDE. However, I don't see any option to let the Buf CLI download the dependencies, in order for this Protobuf plugin to use them (i.e. resolve import paths).
Is that currently supported? The built-in Protobuf plugin by Jetbrains combined with the [official Buf plugin](https://github.com/bufbuild/intellij-buf) does support this, which is great and reduces the complexity of the workflow of working with protocol buffers. However, I'd like to use your Protobuf plugin + Buf dependency management.

Hope to hear a reply soon! Thanks,
Robin

**Comment by trietsch** [comment]
I'm seeing this in the Buf command window. Following the instructions given in the screenshot below has no effect.
<img width="633" alt="image" src="https://github.com/devkanro/intellij-protobuf-plugin/assets/22979744/ac2a5b11-0a72-4345-9baa-760f612e580e">

Looking at the `.idea/protobuf.xml`, the `component/libraries/module` section is not updated (and also mentions incorrect dependencies). While the `component/modules/module` section is correct.

**Reply by trietsch** [reply]
@devkanro any idea?

**Reply by devkanro** [reply]
Can you provide a sample project? it can help me debug

**Reply by devkanro** [reply]
@trietsch

**Reply by trietsch** [reply]
Thanks for your reply! Yes of course. The project I'm working on can be found here: https://github.com/getstrm/pace
Protos can be found in the corresponding dir, other directories do not matter :). Let me know if I need to explain anything.

**Reply by devkanro** [reply]
@trietsch  I found the bug, buf has been updated their cache dir struct, I need update it to v2

**Comment by devkanro** [comment, marked as answer]
@trietsch 
I'm planning to stop maintaining Buf support in my plugin, but I'm modifying the official Buf plugin to support both my plugin and the official Protobuf plugin. You can try the SNAPSHOT version here

[intellij-buf-0.2.2-EAP.zip](https://github.com/devkanro/intellij-protobuf-plugin/files/13726023/intellij-buf-0.2.2-EAP.zip)

**Reply by trietsch** [reply]
Ah cool, so that version of the Buf plugin does not require the Protobuf and gRPC plugins to be enabled in IntelliJ?

**Reply by devkanro** [reply]
Yes, just enable one of my plugin or official plugin, it will work fine.

**Reply by devkanro** [reply]
if the buf plugin not accept my PR, I will create a folk plugin for it.

**Reply by devkanro** [reply]
@trietsch https://github.com/bufbuild/intellij-buf/pull/166 Check the PR state

**Reply by trietsch** [reply]
Awesome, works like a charm! Hope to see https://github.com/bufbuild/intellij-buf/pull/166 merged soon. Thanks for your efforts.

**Reply by iyear** [reply]
Excellent plugin:) The above file requires GoLand 233 and above. You can download plugin with lower version requirements from here(Artifacts): https://github.com/bufbuild/intellij-buf/actions/runs/7458434060?pr=166 and looking forward to the official release of the new version of buf soon.

Backup: 
[intellij-buf-0.2.1.zip](https://github.com/devkanro/intellij-protobuf-plugin/files/13931890/intellij-buf-0.2.1.zip)

**Reply by trietsch** [reply]
Yeah, the response from Buf [looks good](https://github.com/bufbuild/intellij-buf/pull/166#pullrequestreview-1813796988). Hope they'll merge it soon.

**Reply by nicksnyder** [reply]
It is merged and released :)

**Reply by trietsch** [reply]
Thanks @nicksnyder and @devkanro!
